### ISSUE TITLE
{{{ in Mustache Template.

### ISSUE THREAD

**Issue by rileymat** [janl/mustache.js#430]
If I have an object `{"a":"1", "b":"1"}`

And I have a template that I want to use to output a json string
`{"Object1":"whatever",
"nestedObject":{{{#a}}"a":
"{{a}}"{{/a}}
{{^a}}
"b": "{{b}}"
{{/a}}}}`

My output is `{"Object1":"whatever", "nestedObject":}`

If I add a space after `nestedObject{` like:
`{"Object1":"whatever",
"nestedObject":{ {{#a}}"a":
"{{a}}"{{/a}}
{{^a}}
"b": "{{b}}"
{{/a}}}}`

My output is `{"Object1":"whatever", "nestedObject":{ "a": "1" }}`

The second behavior is what I am expecting in the first case as well, but it does not appear to be.
My question, I am not escaping { in some way so it knows it is not part of a triple stache or is this a bug?

**Comment by rileymat** [comment]
Also, I am not sure if this is helpful or not, but an old version (quite a far back, not sure which version because it is not in the JS file) that we were previously using did work as expected.

**Comment by dasilvacontin** [comment]
There's a way to change the chars '{{' used by mustache, look at the docs. (I'm on the phone right now)

Envoyé de mon iPhone

**Comment by rileymat** [comment]
I appreciate the reply, I did a little more research and it looks like it regressed at commit: eb01be04702388fd3cf98e509d3d14f1f9243f09

**Comment by bobthecow** [comment]
The current behavior is correct. `{{{` indicates the start of an unescaped interpolation tag, and it goes until the first `}}}` it finds. So in your case, that's this:

<code>{"Object1":"whatever", "nestedObject":<em>{{{#a}}"a": "{{a}}"{{/a}} {{^a}} "b": "{{b}}" {{/a}}}</em>}</code>

Since you don't have anything in your rendering context called `#a}}"a": "{{a}}"{{/a}} {{^a}} "b": "{{b}}" {{/a`, it renders that tag as an empty string and you get `{"Object1":"whatever", "nestedObject":}`.

**Comment by rileymat** [comment]
I understand and respect your position, thank you for looking at it. 

I just added it to the issue list because in a previous versions it worked the other way and this makes it hard for some people using old versions to upgrade.

Closing issue, I appreciate your responses.

**Comment by dasilvacontin** [comment]
Could you tell us in which version of mustache.js it used to work? Thanks to you for taking the time to report the potential issue. :)

Envoyé de mon iPhone

**Comment by rileymat** [comment]
No problem, I did a git bisect the other day and came across 2 significant commits.
Before eb01be0 it worked in the way where it would allow {{{ without turing it all into a tag and breaking the output. 

After that commit it introduced a crash.

This crash was fixed in a7a6e74, but it did not return to the same behavior as it had before eb01be0.

But, as another contributor as pointed out, the current behavior is the correct behavior for Mustache.  Additionally, I later ran it against pystache, and get the same output as the current Mustache.js. So, I am guessing in the past we were building our templates relying on a bug/specified behavior (unknown to us).
